Middleware/analyzerpcapmiddleware.cpp

```diff
@@ -65,28 +70,27 @@ void AnalyzerPcapMiddleware::bufferProducts() {
             if (startTime == -1) {
                 startTime = ((pcap_pkthdr*)input.header.data())->ts.tv_sec * 1000; // convert to ms
                 startTime += ((pcap_pkthdr*)input.header.data())->ts.tv_usec/1000; // convert to ms
+                lastDuration = 0;
             }
             else {
                 endTime = ((pcap_pkthdr*)input.header.data())->ts.tv_sec*1000;
                 endTime += ((pcap_pkthdr*)input.header.data())->ts.tv_usec/1000; // convert to ms
                 duration = endTime - startTime;
                 if (lastDuration == 0)
                     lastDuration = duration;
-            }
 
-            // 1 second of trafic, calculate bitrate for that second
-
-            if (duration - lastDuration >= 1000) {
-                bitrate = (bytes - lastSecondBytes)*8*1000/(duration - lastDuration);
-                lastSecondBytes = bytes;
-                lastDuration = duration;
+                // 1 second of trafic, calculate bitrate for that second
 
+                if (duration - lastDuration >= emitFrequency) {
+                    bitrate = (bytes - lastSecondBytes)*8*1000/(duration - lastDuration);
+                    lastSecondBytes = bytes;
+     //               lastDuration = duration;
```

Comment:
Don't use lastDuration in two places like this. Not sure this is actually run.

emitFrequency should mean that this is run every emit, in that case just move it to the other "emit" code below.